noirup/noirup

```diff
@@ -60,102 +61,93 @@ main() {
   fi
 
   NOIRUP_REPO=${NOIRUP_REPO-noir-lang/noir}
+  if [[ "$NOIRUP_REPO" == "noir-lang/noir" && -z "$NOIRUP_BRANCH" && -z "$NOIRUP_COMMIT" ]]; then
+    NOIRUP_VERSION=${NOIRUP_VERSION-nightly}
+    NOIRUP_TAG=$NOIRUP_VERSION
+
+    # Normalize versions (handle channels, versions without v prefix)
+    if [[ "$NOIRUP_VERSION" == [[:digit:]]* ]]; then
+      # Add v prefix
+      NOIRUP_VERSION="v${NOIRUP_VERSION}"
+      NOIRUP_TAG="${NOIRUP_VERSION}"
+    fi
+
+    say "installing nargo (version ${NOIRUP_VERSION}, tag ${NOIRUP_TAG})"
+
+    PLATFORM="$(uname -s)"
+    case $PLATFORM in
+      Linux)
+        PLATFORM="unknown-linux-gnu"
+        ;;
+      Darwin)
+        PLATFORM="apple-darwin"
+        ;;
+      *)
+        err "unsupported platform: $PLATFORM"
+        ;;
+    esac
+
+    # We currently only provide binaries targetting x64_64 architectures.
+
+    ARCHITECTURE="x86_64"
```

Comment:
Can we give an error message if the user is not using an x86_64 system? I'd rather an explicit error here than a more confusing one later on.